tests/integ_tests/workflow_tests/replay.rs

```diff
@@ -0,0 +1,86 @@
+use assert_matches::assert_matches;
+use std::time::Duration;
+use temporal_sdk_core::ServerGatewayApis;
+use temporal_sdk_core_api::errors::{PollActivityError, PollWfError};
+use temporal_sdk_core_protos::{
+    coresdk::{workflow_commands::StartTimer, workflow_completion::WorkflowActivationCompletion},
+    temporal::api::workflowservice::v1::PollWorkflowTaskQueueResponse,
+};
+use temporal_sdk_core_test_utils::{
+    history_from_proto_binary, history_replay::mock_gateway_from_history,
+    init_core_replay_preloaded, CoreTestHelpers,
+};
+use tokio::join;
+
+#[tokio::test]
+async fn timer_workflow_replay() {
+    let (core, task_q) = init_core_replay_preloaded(
+        "fail_wf_task",
+        &history_from_proto_binary("histories/fail_wf_task.bin")
```

Comment:
Should we use a history with a more appropriate name?
This is a bit misleading.

Reply:
Heh, I had a history file but wasn't using it since both histories worked with what this test did.